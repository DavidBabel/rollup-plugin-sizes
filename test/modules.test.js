"use strict";

var rollup  = require("rollup").rollup,
    resolve = require("rollup-plugin-node-resolve"),
    
    plugin = require("../index.js"),
    
    // To silence rollup warnings
    external = [
        "fs",
        "path",
        "events",
        "module"
    ];

describe("rollup-plugin-sizes", () => {
    describe("node_modules support", () => {
        var { log, out } = require("./log.js")();

        it("should show basic output", () =>
            rollup({
                input : "./test/specimens/modules/a.js",

                external,

                plugins : [
                    resolve(),
                    plugin({
                        log
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );

        it("should show detailed output ", () =>
            rollup({
                input : "./test/specimens/modules/a.js",

                external,

                plugins : [
                    resolve(),
                    plugin({
                        log,

                        details : true
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );
    });
});
