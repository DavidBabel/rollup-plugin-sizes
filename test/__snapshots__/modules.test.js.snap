// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-sizes node_modules support should show basic output 1`] = `
"Details for \\"bundle:\\"
	rollup - 766.36 KB (99.99%)
	bundle - 58 B (0.01%)
"
`;

exports[`rollup-plugin-sizes node_modules support should show detailed output  1`] = `
"Details for \\"bundle:\\"
	rollup - 766.36 KB (99.99%)
		dist/rollup.es.js - 766.36 KB (100.00%)
	bundle - 58 B (0.01%)
		a.js - 58 B (100.00%)
"
`;
